### PR TITLE
refactor(util): change Position type from string literals to numeric

### PR DESCRIPTION
**Pull request by wittjosiah**
## Summary

- `Position` type in `@dxos/util` changed from `'first' | 'last'` to `number`
- Added `Position` namespace with `Position.first = -Infinity` and `Position.last = Infinity`
- `byPosition` comparator updated to sort numerically, treating `undefined` as `0` (sorts between first and last)
- All 86 call sites across 44 files migrated from `position: 'first'`/`position: 'last'` to `Position.first`/`Position.last`

## Motivation

The two-value union was too coarse for expressing ordering tiers. A numeric type lets callers assign any integer value to slot items precisely between the first/last anchors — e.g. `position: -1` for "just before normal", `position: 1` for "just after normal". The `undefined` (unspecified) case continues to sort at `0`, unchanged.

The `Position.first = -Infinity` / `Position.last = Infinity` constants are drop-in replacements for the old literals and preserve the existing three-tier structure (first → static → last).

## Implementation notes

- `byPosition` returns `-1 | 0 | 1` explicitly to remain compatible with Effect's `Order<T>` type used in `Array.sortBy` calls across `app-graph` and `plugin-deck`.
- `TypeSection.ts` had an inline `position?: 'first' | 'last'` type that was also updated to `Position`.
- No behaviour change for existing code — all existing `'first'` / `'last'` usages map to the same extremes.

## Test plan

- [x] `moon run util:test` — all 104 tests pass, including a new tier-ordering test
- [x] `moon exec --on-failure continue --quiet :build` — zero TypeScript errors across the full repo

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

* **Refactor**
  * Standardized UI/surface/companion and graph ordering across the platform by using shared `Position` constants (`first`/`last`) instead of string literals.
  * Updated exported surface/extension position types to align with the new `Position` representation.
* **Bug Fixes**
  * Improved ordering logic: positions are now compared via numeric priority, with missing positions placed in a predictable middle tier; selection of the “first” candidate now follows the updated rules.
* **Tests**
  * Updated and expanded position-related tests to cover the new `Position` ordering, edge cases, and stability.

### CODE REVIEW

File: packages/sdk/app-graph/src/graph-builder.ts [hunk 1]
```diff
@@ -199,7 +199,10 @@ class GraphBuilderImpl implements GraphBuilder {
     if (ids.length > 0) {
       const sortedIds = [...nodes]
         .sort((a, b) =>
-          byPosition(a.properties ?? ({} as { position?: Position }), b.properties ?? ({} as { position?: Position })),
+          Position.compare(
+            a.properties ?? ({} as { position?: Position.Position }),
+            b.properties ?? ({} as { position?: Position.Position }),
+          ),
```
[the remark this answers is not in the document]
Comment (wittjosiah) [reply packages/sdk/app-graph/src/graph-builder.ts]: Fixed in 9a7823874e: replaced `a.properties ?? ({} as { position?: Position.Position })` with `{ position: a.properties?.position }` — reads position directly from the optional properties chain, no cast needed.

**Comment by wittjosiah** [comment]
Also fixed the pre-existing `args[0] as string` cast in `AppNode.ts` (9a7823874e) — replaced with `String(args[0])` which is type-safe without a cast. The cast was in code touched by this PR but predated it.
